httpcore5/src/test/java/org/apache/hc/core5/concurrent/TestComplexFuture.java

```diff
@@ -96,4 +96,11 @@ public void completed(final Object result) {
         assertThat(dependency2.isCancelled(), CoreMatchers.is(true));
     }
 
+    @Test
+    public void testFailedAndFailed() {
```

Comment:
No strong preference, but we might prefer naming this more specifically `testCanceledAndFailed`